src/py42/clients/trustedactivities.py

```diff
@@ -0,0 +1,83 @@
+from py42.choices import Choices
+
+
+class TrustedActivityType(Choices):
+    DOMAIN = "DOMAIN"
+    SLACK = "SLACK"
+
+
+class TrustedActivitiesClient:
+    """A client to expose the trusted activities/data preferences API
+
+    `Rest documentation <https://developer.code42.com/api/#tag/Trusted-Activities>`__
+    """
+
+    def __init__(self, trusted_activities_service):
+        self._trusted_activities_service = trusted_activities_service
+
+    def get_all(self, type=None):
+        """Gets all trusted activities.
+        `Rest documentation <https://developer.code42.com/>`
+
+        Args:
+            type (str, optional): Type of the trusted activity. `TrustedActivityType.DOMAIN` or `TrustedActivityType.SLACK`
+
+        Returns:
+            :class:'py42.response.Py42Response'
+        """
+        return self._trusted_activities_service.get_all(type)
+
+    def create(self, type, value, description=None):
+        """Gets all trusted activities with the given type.
+        `Rest documentation <https://developer.code42.com/>`
+
+        Args:
+            type (str): Type of the trusted activity. `TrustedActivityType.DOMAIN` or `TrustedActivityType.SLACK`
+            value (str): The URL of the domain or name of the Slack workspace.
+            description (str, optional): Description of the trusted activity.
+
+        Returns:
+            :class:'py42.response.Py42Response'
+        """
+        return self._trusted_activities_service.create(type, value, description)
+
+    def get(self, id):
+        """Retrieve trusted activity details by given resource number.
+        `Rest documentation <https://developer.code42.com/>`
+
+        Args:
+            id (int): Resource number of the trusted activity or domain.
+
+        Returns:
+            :class:'py42.response.Py42Response'
+        """
+        return self._trusted_activities_service.get(id)
+
+    def update(self, id, type=None, value=None, description=None):
+        """Updates trusted activity details by given resource number.
+        `Rest documentation <https://developer.code42.com/>`
+
+        Args:
+            id (int): Resource number of the trusted activity.
+            type (str, optional): Type of the trusted activity. `TrustedActivityType.DOMAIN` or `TrustedActivityType.SLACK`
+            value (str, optional): The URL of the domain or name of the Slack workspace.
+            description (str, optional): Description of the trusted activity.
+
+        Returns:
+            :class:'py42.response.Py42Response'
+        """
+        return self._trusted_activities_service.update(
```

Comment:
A side effect of the way the update API works is that it will replace all properties with whatever was sent -- i.e. if you only want to update the value and don't send a description, then the description will be updated to be blank.

To avoid this oddity, we can first get the activity that we want to update and build the update request based off of that. Here's an example of how we manage that problem in the cases API:

https://github.com/code42/py42/blob/0d0a472bfce6d1ccd5ae09eb5789e8494c0da247/src/py42/services/cases.py#L110

Reply:
the cases API does this in its services method (rather than the client) - I did this same thing for the trustedactivities `update()` method its just in `services`

Reply:
oh i missed that thanks, nice!